mofa/model.py

```diff
@@ -130,13 +176,52 @@ class LigandDescription:
     dummy_element: str = field(default=None, repr=False)
     """Element used to represent the end group during assembly"""
 
+
+
     @cached_property
     def atoms(self):
         return read_from_string(self.xyz, "xyz")
 
-    def replace_with_dummy_atoms(self) -> ase.Atoms:
+    def replace_with_dummy_atoms(self, anchor_types: str="COO") -> ase.Atoms:
         """Replace the fragments which attach to nodes with dummy atoms"""
-        raise NotImplementedError()
+        rdmol = Chem.MolFromSmiles(self.smiles)
+        rdmol = Chem.AddHs(rdmol)
+        AllChem.EmbedMolecule(rdmol)
+        AllChem.UFFOptimizeMolecule(rdmol)
+        old_Natoms = rdmol.GetNumAtoms() + 1
+        curr_Natoms = rdmol.GetNumAtoms()
+        while old_Natoms != curr_Natoms:
+            dummyElement = "At"
+            dummyAtomicNum = Chem.rdchem.Atom(dummyElement).GetAtomicNum()
+            match_ids = rdmol.GetSubstructMatch(Chem.MolFromSmiles("C(=O)O"))
+            carboxylicC = [at for at in match_ids if rdmol.GetAtomWithIdx(at).GetSymbol() == "C"]
+            if len(carboxylicC) > 0:
+                carboxylicC = carboxylicC[0]
+            else:
+                break
+            bonds2rm = [tuple(sorted((at, carboxylicC)))
+                        for at in match_ids if rdmol.GetAtomWithIdx(at).GetSymbol() == "O"]
+            rdmol.GetAtomWithIdx(carboxylicC).SetAtomicNum(dummyAtomicNum)
```

Comment:
Could I avoid all this logic for finding which atom to remove if we mark the correct atom in the template description?